src/storage-client/src/storage_collections.rs

```diff
@@ -1061,23 +1061,23 @@ where
         }
 
         // Translate our net compute actions into downgrades of persist sinces.
-        // The actual downgrades are performed by a Tokio task asynchorously.
-        let mut persist_compaction_commands = BTreeMap::default();
+        // The actual downgrades are performed by a Tokio task asynchronously.
+        let mut persist_compaction_commands = Vec::with_capacity(collections_net.len());
```

Comment:
Can you say more to the optimization? ISTM that whether it is an optimization or a pessimization depends on the changes in `collections_net`. If there are no changes, the previous code wouldn't allocate, but with this change we always allocate once. Or am I missing something?